matplotcheck/tests/test_base_data.py

```diff
@@ -0,0 +1,87 @@
+"""Tests for the base module -- Data"""
+import pytest
+import matplotlib.pyplot as plt
+import numpy as np
+import random
+
+
+"""DATACHECK TESTS"""
+
+
+def test_assert_xydata_scatter(pt_scatter_plt, pd_df):
+    """Checks points in scatter plot against expected data"""
+    pt_scatter_plt.assert_xydata(pd_df, xcol="A", ycol="B")
+    plt.close()
+
+
+def test_assert_xydata_tolerance(pt_scatter_plt, pd_df):
+    """Checks that slightly altered data still passes with an appropriate
+    tolerance"""
+    for i in range(len(pd_df["A"])):
+        pd_df["A"][i] = pd_df["A"][i] + (np.floor(pd_df["A"][i] * 0.25))
+        pd_df["B"][i] = pd_df["B"][i] + (np.floor(pd_df["B"][i] * 0.25))
+    pt_scatter_plt.assert_xydata(pd_df, xcol="A", ycol="B", tolerence=0.5)
+    plt.close()
+
+
+def test_assert_xydata_tolerance_fail(pt_scatter_plt, pd_df):
+    """Checks that data altered beyond the tolerance throws an assertion."""
+    pd_df["A"][1] = pd_df["A"][1] * 2
+    with pytest.raises(AssertionError, match="Incorrect data values"):
+        pt_scatter_plt.assert_xydata(pd_df, xcol="A", ycol="B", tolerence=0.1)
+    plt.close()
+
+
+def test_assert_xydata_changed_data(pt_scatter_plt, pd_df):
+    """assert_xydata should fail when we change the data"""
+    pd_df["B"][1] += 5
+    with pytest.raises(AssertionError, match="Incorrect data values"):
+        pt_scatter_plt.assert_xydata(pd_df, xcol="A", ycol="B")
+    plt.close()
+
+
+def test_assert_xydata_scatter_points_only(pt_scatter_plt, pd_df):
+    """Checks points in scatter plot against expected data"""
+    pt_scatter_plt.assert_xydata(pd_df, xcol="A", ycol="B", points_only=True)
+    plt.close()
+
+
+def test_assert_xydata_changed_data_points_only(pt_scatter_plt, pd_df):
+    """assert_xydata should fail when we change the data"""
+    pd_df["B"][1] += 5
+    with pytest.raises(AssertionError, match="Incorrect data values"):
+        pt_scatter_plt.assert_xydata(
+            pd_df, xcol="A", ycol="B", points_only=True
+        )
+    plt.close()
+
+
+'''
```

Comment:
@ryla5068 super minor. will you please move this code OUT of the test_base_data.py file and create a new test_base_timeseries_data.py file? it's ok if it remains commented out id just like it in a new file.